nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -951,7 +951,7 @@ def wait_for_x(self) -> None:
         def check_x(_: Any) -> bool:
             cmd = (
                 "journalctl -b SYSLOG_IDENTIFIER=systemd | "
-                + 'grep "Reached target Current graphical"'
+                + 'grep "Reached target graphical-session.target"'
```

Comment:
huh, systemd should use the unit name from v244-ish which is.... a while ago (2019)

That can be configured through e.g. `StatusUnitFormat=description` (or name) in system.conf but we don't seem to touch that, I wonder why nixos-unstable-small would use the unit name there....

Reply:
What would cause the specific combination of 252 on nixos-unstable-small to use names? 251 on *either* branch and 252 on staging uses descriptions.

Reply:
I have a hunch that it's #201177, which is on staging but not nixos-unstable-small. I bet that without that, in more recent systemd versions, it uses names instead of descriptions by default.

Reply:
It's this commit: https://github.com/systemd/systemd/commit/819c0dcbf510dab8044e77e98e7593b384ad908c
combined with our commit in nixpkgs that builds systemd in release mode that you probably don't have yet nixos-unstable-small which makes it default to building developer mode

sneaky change not in NEWS, but I guess it only impacts developers (... and distros that forget to explicitly build release mode?), but I guess at least we know why...

Reply:
Yea I'll build on nixos-unstable-small *just to be sure*, but it looks like this change should be dropped from this PR.

Reply:
Confirmed.

Reply:
But does this mean this should be fixed in staging, as a followup for https://github.com/NixOS/nixpkgs/pull/201177 ?

Reply:
Ah `nixos/lib/test-driver/test_driver/machine.py` matched on the unit description all the time. nvm.